rskj-core/build.gradle

```diff
@@ -189,18 +243,31 @@ task generatePom(dependsOn: jar) {
 }
 
 task fatJar(type: Jar, dependsOn: jar) {
+    archiveClassifier.set('all')
+    duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
     manifest {
         attributes 'Main-Class': "$mainClassName"
     }
-    exclude "META-INF/*.SF", "META-INF/*.DSA", "META-INF/*.RSA"
-    classifier = 'all'
+
+    exclude 'META-INF/*.SF', 'META-INF/*.DSA', 'META-INF/*.RSA'
+
+    exclude 'META-INF/NOTICE'
+    exclude 'META-INF/NOTICE.md'
+    exclude 'META-INF/NOTICE.txt'
+    exclude 'META-INF/LICENSE'
+    exclude 'META-INF/LICENSE.md'
+    exclude 'META-INF/LICENSE.txt'
```

Comment:
We need legal counsul before getting rid of it. Also I'd add a comment stating that we're excluding those files because it generates weird differences in the reproducible build, thing that bogs me a lot. Do we have any clue why this is happening? I haven't found anything yet, sadly.

Reply:
>using the .set() method for the deprecated variables feels off compared to the rest of the file. IIRC, archiveClassifier = 'sources' instead of archiveClassifier.ste('sources') should work too, and would prefer it like that.

Yeah, I'd also prefer this, but I did it due to the annoying message of my IDE - `Cannot assign 'String' to 'Property<String>' `

>We need legal counsul before getting rid of it. 

Agree, created this as a draft just to open the discussion

>Also I'd add a comment stating that we're excluding those files because it generates weird differences in the reproducible build, thing that bogs me a lot. Do we have any clue why this is happening?

The thing is that multiple third-party jars are being merged into the final uber/fat jar and some file names of them may clash. 

Before this change it was done the following way: first, gradle `fatJar` task puts all files from external dependencies into the `*-all.jar` file including duplicates of `.class`, `LICENSE`/`LICENSE.*`, `NOTICE`/`NOTICE.*` etc. files. After that, `stripJar` task removes duplicate entries from the jar keeping only first file with unique name and re-sets timestamps (which is useless in fact, cause `preserveFileTimestamps` flag does its job beforehand). As duplicates are removed, as of now, not all `LICENSE` and `NOTICE` files are being included into the final jar, only one instance.

My change removes the `stripJar` part and sets `EXCLUDE` duplicate strategy for the gradle task via this `duplicatesStrategy = DuplicatesStrategy.EXCLUDE` (`INCLUDE` is by default), which simply removes duplicate entries on its own. Also the change excludes `LICENSE`/`LICENSE.*`, `NOTICE`/`NOTICE.*` files, which is a question to discuss.

Regarding `module-info.class`, there're also a few entries of it from others deps. As of now, only one instance of them goes into the final jar. This file is related to the new java module system. As we provide an app, not a lib, I believe we could also get rid of it - OFC needs to be properly tested.